motionplan/motionPlanner.go

```diff
@@ -381,3 +383,92 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan Plan,
+	worldState *frame.WorldState,
+	fs frame.FrameSystem,
+	currentPosition spatialmath.Pose,
+	errorState spatialmath.Pose,
+	logger *zap.SugaredLogger,
+) error {
+	// ensure that we can actually perform the check
+	if len(plan) < 1 {
+		return errors.New("plan must have at least one element")
+	}
+
+	// construct solverFrame
+	// Note that this requires all frames which move as part of the plan, to have an
+	// entry in the very first plan waypoint
+	sf, err := newSolverFrame(fs, checkFrame.Name(), frame.World, plan[0])
+	if err != nil {
+		return err
+	}
+
+	// construct planager
+	sfPlanner, err := newPlanManager(sf, fs, logger, defaultRandomSeed)
```

Comment:
RRT algorithms are random by design. The philosophy of randomly sampling a space rather than exhaustively searching it lends itself to better performance in high dimensional spaces